python/CLOUDTRAIL_S3_DATAEVENTS_ENABLED/CLOUDTRAIL_S3_DATAEVENTS_ENABLED.py

```diff
@@ -0,0 +1,489 @@
+    if not trail_list:
+        evaluations.append(build_evaluation(event['accountId'], 'NON_COMPLIANT', event, DEFAULT_RESOURCE_TYPE, 'There are no AWS CloudTrail Trails in the region.'))
+        return evaluations
+
+    #No rule parameter specified, default check for all S3 buckets
+    if not valid_rule_parameters:
+        for trail in trail_list:
+            if trail['HasCustomEventSelectors']:
+                #example arn: arn:aws:cloudtrail:us-east-1:123456789012:trail/test
+                region = trail['TrailARN'][19:].split(':')[0]
+                ct_regional_client = boto3.client('cloudtrail', region_name=region)
+                event_selector_output = ct_regional_client.get_event_selectors(TrailName=trail['Name'])['EventSelectors']
+                data_resource_values = get_data_resource_s3_values(event_selector_output)
+                if isinstance(data_resource_values, bool):
+                    evaluations.append(build_evaluation(event['accountId'], 'COMPLIANT', event, DEFAULT_RESOURCE_TYPE))
+                    return evaluations
+
+    #Rule parameter S3BucketName is specified
+    else:
+        for trail in trail_list:
+            if trail['HasCustomEventSelectors']:
+                #example arn: arn:aws:cloudtrail:us-east-1:123456789012:trail/test
+                region = trail['TrailARN'][19:].split(':')[0]
+                ct_regional_client = boto3.client('cloudtrail', region_name=region)
+                event_selector_output = ct_regional_client.get_event_selectors(TrailName=trail['Name'])['EventSelectors']
+                data_resource_values = get_data_resource_s3_values(event_selector_output)
+                #if logging is enabled for 'arn:aws:s3', return COMPLIANT
+                if isinstance(data_resource_values, bool):
+                    evaluations.append(build_evaluation(event['accountId'], 'COMPLIANT', event, DEFAULT_RESOURCE_TYPE))
+                    return evaluations
+
+                #get bucket names from data resource values to store them in a list
+                for value in data_resource_values:
+                    # value is of the form arn:aws:s3:::test/, [13:-1] strips 'arn:aws:s3:::' and the trailing slash
+                    bucket_name = value[13:-1]
+                    compliant_buckets.append(bucket_name)
+
+        #match rule parameter values v/s buckets with data events logging enabled
+        non_compliant_buckets = (list(set(valid_rule_parameters) - set(compliant_buckets)))
+        #If non_compliant_buckets is empty, logging is enabled for all required buckets
+        if not non_compliant_buckets:
+            evaluations.append(build_evaluation(event['accountId'], 'COMPLIANT', event, DEFAULT_RESOURCE_TYPE))
+            return evaluations
+
+        evaluations.append(build_evaluation(event['accountId'], 'NON_COMPLIANT', event, DEFAULT_RESOURCE_TYPE, 'Not valid for some buckets: ' + str(non_compliant_buckets) + '.'))

```

Comment:
Change annotation to-'AWS CloudTrail trails do not log S3 data events for buckets: ' + str(non_compliant_buckets) + '.'